File: server/Services/VaultKeepService.cs

namespace keepr.Services;

public class VaultKeepService
{
    private readonly VaultKeepRepository _repository;

    public VaultKeepService(VaultKeepRepository repository)
    {
        _repository = repository;
    }

    internal VaultKeep CreateVaultKeep(VaultKeep vaultkeepData, string userId)
    {
        vaultkeepData.CreatorId = userId;
        VaultKeep vaultkeep = _repository.CreateVaultKeep(vaultkeepData);
        return vaultkeep;
    }

    internal VaultKeep GetVaultKeepById(int vaultkeepId)
    {
        VaultKeep vaultkeep = _repository.GetVaultKeepById(vaultkeepId);
        if (vaultkeep == null)
        {
            throw new Exception("Unable to find vaultkeep!");
        }
        return vaultkeep;
    }

    internal List<VaultKeep> GetVaultKeeps(string userId)
    {
        List<VaultKeep> vaultkeeps = _repository.GetVaultKeeps(userId);
        return vaultkeeps;
    }

    internal string TrashVaultKeep(int vaultkeepId, string userId)
    {
        VaultKeep foundVK = GetVaultKeepById(vaultkeepId);
        if (foundVK.CreatorId != userId)
        {
            throw new Exception("You cannot delete what isn't yours!");
        }
        _repository.TrashVaultKeep(vaultkeepId);
        return "VaultKeep deleted!";
    }
}

File: client/src/services/VaultKeepService.js
import { AppState } from "../AppState.js"
import { VaultKeep } from "../models/VaultKeep.js"
import { api } from "./AxiosService.js"


class VaultKeepService{
    async getVaultKeepsByVault(vaultId){
        const response = await api.get(`api/vault/${vaultId}/vaultkeeps`)
        const vaultkeeps = response.data.map(vaultkeep => new VaultKeep(vaultkeep))
        AppState.keepsInVault.push(vaultkeeps)
    }

    async createVaultKeep(vaultkeepData){
        vaultkeepData.creatorId = AppState.account.id
        const response = await api.post("api/vaultkeeps", vaultkeepData)
        const vaultkeep = new VaultKeep(response.data)
        AppState.vaultkeeps.push(vaultkeep)
    }

    async trashVaultKeep(vaultkeepId){
        await api.delete(`api/vaultkeeps/${vaultkeepId}`)
        const trash = AppState.keepsInVault.findIndex(vaultkeep => vaultkeep.vaultKeepId = vaultkeepId)
        AppState.vaultkeeps.splice(trash, 1)
    }

    async getVaultKeeps(){
        const response = await api.get("api/vaultkeeps")
        const vaultkeeps = response.data.map(vaultkeep => new VaultKeep(vaultkeep))
        AppState.vaultkeeps = vaultkeeps
    }

    async getVaultKeepById(vaultkeepId){
        const response = await api.get(`api/vaultkeeps/${vaultkeepId}`)
        const vaultkeep = new VaultKeep(response.data)
        AppState.activeVaultKeep = vaultkeep
    }

}

export const vaultKeepService = new VaultKeepService()